pyk/src/pyk/tests/test_kcfg.py

```diff
@@ -263,11 +258,75 @@ def test_paths_between(self):
             },
         )
 
+    def test_resolve(self):
+        # Given
+        d = {
+            'nodes': node_dicts(4),
+            'edges': edge_dicts((0, 1), (0, 2), (1, 2), (1, 3), (2, 3), (3, 0)),
+        }
+        cfg = KCFG.from_dict(d)
+
+        self.assertEqual(node(1), cfg.node('d33...d8'))
+        self.assertEqual(node(1), cfg.node('d33..d8'))
+        self.assertEqual(node(1), cfg.node(node(1).id))
+
+        # Matches no nodes
+        with self.assertRaisesRegex(ValueError, 'Unknown node: deadbeef\\.\\.\\.d8'):
+            self.assertEqual(node(1), cfg.node('deadbeef...d8'))
+
+        # Bad short hash: Has digits between dots
+        with self.assertRaisesRegex(ValueError, 'Bad short hash: 3\\.c62e73544\\.\\.\\.'):
+            cfg.node('3.c62e73544...')
```

Comment:
These are not really `KCFG` tests, so they should be moved to their own module (e.g. `test_hash`).